src/modules/job-manager/event.c

```diff
@@ -521,14 +521,15 @@ static int get_timestamp_now (double *timestamp)
     return 0;
 }
 
-/* we need to send the job id along with each eventlog entry, so wrap
- * the eventlog entry in another object with the job id
+/* we need to send the job id & seq along with each eventlog entry, so
+ * wrap the eventlog entry in another object with the job id & seq
  */
```

Comment:
Sounds good.  FYI per discussion in #3209 

```
Better would be to include a per-job event sequence number in the journal event envelope (where we are adding the job id). During KVS replay, both producer (job manager) and consumer would count how many events are applied to each job. Then, for each job, events in the journal could be discarded if they contain an earlier sequence number. This would also provide a way to detect if events were missed, e.g. if the journal buffer was overrun at the source.
```

Reply:
If that's the case don't we need to increment job->seq in restart.c?  (Or rather `job_create_from_eventlog()` in job.c?)  Your comment reminded me that the main use case for this is when both job-info and job-manager are restarting and are creating initial state from the KVS.  There could be some synchronization issues there at start up.

Also would be good in that comment to say something about the fact that the sequence does not count JOURNAL_ONLY events.

Reply:
> If that's the case don't we need to increment job->seq in restart.c?

Oh crud!  You're right, I had totally forgotten that.  I did it in the `job-info` side (follow up PR), but not on the `job-manager` side.